### PR TITLE
(PC-11846) feat(TMX): Intégration de TMX [WEB]

### PR DESCRIPTION
**Pull request by kopax-polyconseil**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-11846

## Checklist

I have:

- [x] Made sure the title of my PR follows the [convention](../doc/standards/pr-title.md) `($jira) $type($scope): $summary`.
- [x] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [ ] Written **unit tests** native (and web when implementation is different) for my feature.
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** or **github dev name** for any added TODO / FIXME.
- [ ] Added new reusable components to **AppComponents** page and communicate to the team on slack.
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" : `https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a`
- [ ] Made sure translations file is up to date (`yarn translations:extract`)

## Screenshots

| iOS | Android | Mobile - Chrome | Desktop - Chrome | Desktop - Safari |
| --: | ------: | --------------: | ---------------: | ---------------: |
|     |         |                 |                  |                  |

### CODE REVIEW

File: src/libs/profiling/index.web.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  profileDevice(orgId: string, fpServer: string, callback: (sessionId: string) => void): void
+}
+
+const profileDevice = async (
+  orgId: string,
+  fpServer: string,
+  callback: (sessionId: string) => void
+) => {
+  api
+    .getnativev1userProfilingsessionId()
```
Comment (antoinewg) [inline src/libs/profiling/index.web.ts:17]: tu le fais via le back finalement ? 👍

**Review by antoinewg — COMMENTED** [review]
lgtm

File: src/libs/profiling/index.web.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    .then(({ sessionId }: UserProfilingSessionIdResponse) => {
+      if (!sessionId) {
+        new MonitoringError(
+          'GET /native/v1/user/profiling/session_id did not return a sessionId: profiling skipped'
```
Comment (prouzet) [inline src/libs/profiling/index.web.ts:21]: user_profiling et non user/profiling, mais ici ce n'est qu'un log

Comment (kopax-polyconseil) [reply src/libs/profiling/index.web.ts:21]: Merci je corrige

File: src/libs/profiling/index.web.ts (hunk 1 above)
Comment (kopax-polyconseil) [reply src/libs/profiling/index.web.ts:17]: rep via MP

File: src/features/auth/signup/PhoneValidation/SetPhoneValidationCode.tsx [hunk 3]
```diff
@@ -119,7 +122,11 @@ export const SetPhoneValidationCode = memo(({ route }: SetPhoneValidationCodePro
     } else {
       await api
         .postnativev1userProfiling({
-          session_id: sessionId,
+          agentType: Platform.select({
```
[the remark this answers is not in the document]
Comment (kopax-polyconseil) [reply src/features/auth/signup/PhoneValidation/SetPhoneValidationCode.tsx:125]: Si c'est possible, je ne l'ai juste pas fais, c'est pas ça qui causera une fuite mémoire.

**Review by antoinewg — APPROVED** [review]
modulo le bump de version react-native-profiling

**Pull request merged**